src/creature_states.c

```diff
@@ -3044,42 +3042,86 @@ short creature_wants_a_home(struct Thing *creatng)
     return 0;
 }
 
-short creature_wants_salary(struct Thing *creatng)
+struct Room* get_room_for_thing_salary(struct Thing* creatng, unsigned char *navtype)
 {
-    SYNCDBG(8,"Starting for %s index %d owner %d", thing_model_name(creatng),(int)creatng->index,(int)creatng->owner);
     TRACE_THING(creatng);
-    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
-    struct Room* room = find_nearest_room_for_thing_with_used_capacity(creatng, creatng->owner, RoK_TREASURE, NavRtF_Default, 1);
+    RoomKind job_rkind = get_room_for_job(Job_TAKE_SALARY);
+    if (!player_has_room(creatng->owner, job_rkind))
+    {
+        return INVALID_ROOM;
+    }
+        
+    //struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
```

Comment:
any reason to leave this commented out code in? it got moved to the other function right?

Reply:
Just forgot to clean it up, removed.